arelle/XmlUtil.py

```diff
@@ -377,24 +501,43 @@ def childrenAttrs(element, childNamespaceURI, childLocalNames, attrLocalName):
     childrenAttrs.sort()
     return childrenAttrs
 
-def descendant(element, descendantNamespaceURI, descendantLocalNames, attrName=None, attrValue=None) -> Any:
+def descendant(
+    element: ModelObject,
+    descendantNamespaceURI: str,
+    descendantLocalNames: str | tuple[str, ...],
+    attrName: str | None = None,
+    attrValue: str | None = None
+) -> ModelObject | PrototypeObject | None:
     d = descendants(element, descendantNamespaceURI, descendantLocalNames, attrName, attrValue, breakOnFirst=True)
     if d:
         return d[0]
     return None
 
-def descendants(element, descendantNamespaceURI, descendantLocalNames, attrName=None, attrValue=None, breakOnFirst=False, ixTarget=False):
+def descendants(
+    element: ModelObject,
+    descendantNamespaceURI: str,
+    descendantLocalNames: str | tuple[str, ...],
+    attrName: str | None = None,
+    attrValue: str | None = None,
+    breakOnFirst: bool = False,
+    ixTarget: bool = False
+) -> list[ModelObject | PrototypeObject]:
     descendants = []
     if not isinstance(descendantLocalNames,tuple): descendantLocalNames = (descendantLocalNames ,)
     wildLocalName = descendantLocalNames == ('*',)
     wildNamespaceURI = not descendantNamespaceURI or descendantNamespaceURI == '*'
     if isinstance(element,(ModelObject,etree._ElementTree,PrototypeElementTree,PrototypeObject)):
-        for child in (element.ixIter() if ixTarget and hasattr(element, "ixIter") else
+        for child in (getattr(element, 'ixIter')() if ixTarget and hasattr(element, "ixIter") else
                       element.iterdescendants() if isinstance(element,ModelObject) else
                       element.iter()):
+            _childNamespaceURI = getattr(child, 'elementNamespaceURI', None)
+            _childLocalName = getattr(child, 'localName', None)
+            if isinstance(child, ModelObject) and ixTarget:
+                _childNamespaceURI = child.qname.namespaceURI
+                _childLocalName = child.qname.localName
```

Comment:
I'm not sure about this change. The `PrototypeObject` base class doesn't have a qname property, but some (all?) subclasses of it do. It would be safer to keep the prototype isinstance check, but use `getattr` to avoid a potential attribute and mypy error.
```suggestion
            if isinstance(child, (ModelObject, PrototypeObject)) and ixTarget:
                childQname: QName | None = getattr(child, 'qname', None)
                if childQname:
                    _childNamespaceURI = childQname.namespaceURI
                    _childLocalName = childQname.localName
```